src/Neo.IO/ByteArrayComparer.cs

```diff
@@ -20,27 +21,29 @@ internal class ByteArrayComparer : IComparer<byte[]>
         public static readonly ByteArrayComparer Default = new(1);
         public static readonly ByteArrayComparer Reverse = new(-1);
 
-        private readonly int direction;
+        private readonly int _direction;
 
         private ByteArrayComparer(int direction)
         {
-            this.direction = direction;
+            this._direction = direction;
         }
 
-        public int Compare(byte[] x, byte[] y)
+        public int Compare([NotNullWhen(true)] byte[]? x, [NotNullWhen(true)] byte[]? y)
         {
-            return direction > 0
-                ? CompareInternal(x, y)
-                : -CompareInternal(x, y);
+            if (ReferenceEquals(x, y)) return 0;
+
+            return _direction > 0 ?
+                CompareInternal(x!, y!) :
```

Comment:
x or y can be null here

Reply:
Ok thats what i wanted to know. How should `null` be handled? We don't check for `null`....